### ISSUE TITLE
[Submission details] Show geometry annotation reference on map

### ISSUE THREAD

**Issue by amysorto** [google/ground-platform#1477]
Currently after the submission of #1472, the number annotations don't appear on the map but only on the submissions panel.

Mock:
![image](https://github.com/google/ground-platform/assets/8575252/d82227bb-fd86-4dec-895d-f543465b3e3c)

**Comment by gino-m** [comment]
@amysorto @jcqli Let's punt on this one for now?

**Comment by gino-m** [comment]
@amysorto Please be sure numbers are sequential starting at "1". 🙏

**Comment by kenstershiro** [comment]
@amysorto any update on this one?

**Comment by amysorto** [comment]
Closing this issue since this was added to the app for points: #2109.

Another issue is filed for polygons since that is currently blocked by some other work to enable it: https://github.com/google/ground-platform/issues/2126
